func/checkparameters.py

```diff
@@ -1,267 +1,165 @@
   binary = getbinary(gametype)
 
-  #Check if parameters are present (launcherArgs, enviromentOptions, targetExe)  
-  def ifpresent(parameter):
-
-    if parameter in game[appname].keys():
-      return True
-
   try:
     #CONFIGURING BOOLEAN PARAMETERS
 
-    #audioFix
-    audioFix = ""
-    if ifpresent("audioFix"):
-
-      if game[appname]["audioFix"]:
-        audioFix = "PULSE_LATENCY_MSEC=60 "
-
-    #print(audioFix)
-
-
     #Auto-Cloud Save Sync
     cloudsync = ""
-    if ifpresent("autoSyncSaves"):
-
-      if game[appname]["autoSyncSaves"]:
-    
-        if game[appname]["savesPath"] != "":
+    if gameSettings.get("autoSyncSaves") and gameSettings.get("savesPath"):
           #Download and Upload
-          cloudsync = binary + 'sync-saves --save-path "' + game[appname]["savesPath"] + '" ' + appname + ' -y '
-
-    #print(cloudsync)
-
+          cloudsync = f"{binary} sync-saves --save-path \"{gameSettings['savesPath']}\" {appname} -y"
 
-    #enableEsync (Wine, Proton)
-    enableEsync = ["","PROTON_NO_ESYNC=1 "]
-    if ifpresent("enableEsync"):
+    # Set environment variables first
 
-      if game[appname]["enableEsync"]:
-        enableEsync[0] = "WINEESYNC=1 "
-        enableEsync[1] = ""
-
-    #print(enableEsync)
-
-
-    #enableFsync (Wine, Proton)
-    enableFsync = ["", "PROTON_NO_FSYNC=1 "]
-    if ifpresent("enableFsync"):
-
-      if game[appname]["enableFsync"]:
-        enableFsync[0] = "WINEFSYNC=1 "
-        enableFsync[1] = ""
-
-    #print(enableFsync)
-
-
-    #enableFSR & Sharpness
-    enableFSR = ""
-    maxSharpness = ""
-    if ifpresent("enableFSR"):
-
-      if game[appname]["enableFSR"]:
-        enableFSR = "WINE_FULLSCREEN_FSR=1 "
-        maxSharpness = "WINE_FULLSCREEN_FSR_STRENGTH=" + str(game[appname]["maxSharpness"]) + " " 
-
-    #print(enableFSR)
+    #showFps
+    if gameSettings.get("showFps"):
+        environment["DVXK_HUD"] = "fps"
 
     #DXVK FPS Limit
-    dxvk_fps = ""
-    if ifpresent("enableDXVKFpsLimit"):
+    if gameSettings.get("enableDXVKFpsLimit"): 
+        environment["DXVK_FRAME_RATE"] = str(gameSettings.get("DXVKFpsCap")) if gameSettings.get("DXVKFpsCap") else "60"
 
-      try:
-        if game[appname]["enableDXVKFpsLimit"]:
-
-          if game[appname]["DXVKFpsCap"] == "":
-            dxvk_fps = "DXVK_FRAME_RATE=60 "
-          else:
-            dxvk_fps = "DXVK_FRAME_RATE=" + game[appname]["DXVKFpsCap"] + " "
-            
-      except:
-        pass
-
-
-    #enableResizableBar
-    enableResizableBar = ""
-    if ifpresent("enableResizableBar"):
+    #enableFSR & Sharpness
+    if gameSettings.get("enableFSR") and gameSettings.get("maxSharpness"):
+        environment["WINE_FULLSCREEN_FSR"] = "1"
+        environment["WINE_FULLSCREEN_FSR_STRENGTH"] = str(gameSettings.get("maxSharpness"))
 
-      if game[appname]["enableResizableBar"]:
-        enableResizableBar = "VKD3D_CONFIG=upload_hvv "
+    #enableEsync
+    if gameSettings.get("enableEsync"):
```

Comment:
It's better to use this only for wine using `gametype`, since the `PROTON_NO_ESYNC` will be included for GOG Linux games.

Reply:
Good idea. I have made this change and baselined the existing tests with the new output.